xcdat/dataset.py

```diff
@@ -633,22 +650,27 @@ def _get_cftime_coords(
     np.ndarray
         An array of `cftime` coordinates.
     """
-    # Create an array of `datetime` objects by adding the number offset to the
-    # reference date. The `parse.parse` default is set to datetime(2000, 1, 1),
-    # with each component being a placeholder if the value does not exist.
-    ref_datetime: datetime = parser.parse(ref_date, default=datetime(2000, 1, 1))
-
-    # Convert offsets to `np.float64` to avoid "TypeError: unsupported type for
-    # timedelta days component: numpy.int64".
-    offsets = offsets.astype("float64")
-    times = np.array(
-        [ref_datetime + rd.relativedelta(**{units: offset}) for offset in offsets],
-        dtype="object",
-    )
-
-    # Convert the array of `datetime` objects into `cftime` objects based on
-    # the calendar type.
-    date_type = get_date_type(calendar)
-    coords = convert_times(times, date_type=date_type)
+    if units in NON_CF_TIME_UNITS:
+        # Convert offsets to `np.float64` to avoid "TypeError: unsupported type
+        # for timedelta days component: numpy.int64".
+        offsets = offsets.astype("float64")
+
+        # We don't need to do calendar arithmetic here because the units and
+        # offsets are in "months" or "years", which means leap days should not
+        # be factored.
+        ref_datetime: datetime = parser.parse(ref_date, default=datetime(2000, 1, 1))
+        times = np.array(
+            [ref_datetime + rd.relativedelta(**{units: offset}) for offset in offsets],
+            dtype="object",
+        )
+        # Convert the array of `datetime` objects into `cftime` objects based on
+        # the calendar type.
+        date_type = get_date_type(calendar)
+        coords = convert_times(times, date_type=date_type)
+    else:
+        xrunits = units + " since " + ref_date
+        coords = xr.coding.times.decode_cf_datetime(
+            offsets, xrunits, calendar=calendar, use_cftime=True
+        )
```

Comment:
I used your suggested changes.

Reply:
We can probably further increase performance __for CF compliant datasets__ by reverting the CF compliance check in `open_mfdataset()`, then using `decode_times=True` and `use_cftimes=True` as you mentioned. 

I'm taking a look at this right now.